alibi_detect/utils/pytorch/kernels.py

```diff
@@ -88,6 +168,271 @@ def forward(self, x: Union[np.ndarray, torch.Tensor], y: Union[np.ndarray, torch
         return kernel_mat.mean(dim=0)  # [Nx, Ny]
 
 
+class RationalQuadratic(BaseKernel):
+    def __init__(
+        self,
+        alpha: torch.Tensor = None,
+        init_fn_alpha: Callable = pseudo_init_fn,
+        sigma: torch.Tensor = None,
+        init_fn_sigma: Callable = sigma_median,
+        trainable: bool = False,
+        active_dims: Optional[list] = None,
+        feature_axis: int = -1
+    ) -> None:
+        """
+        Rational Quadratic kernel: k(x,y) = (1 + ||x-y||^2 / (2*sigma^2))^(-alpha).
+        A forward pass takesa batch of instances x [Nx, features] and y [Ny, features]
+        and returns the kernel matrix [Nx, Ny].
+
+        Parameters
+        ----------
+        alpha
+            Exponent parameter of the kernel.
+        sigma
+            Bandwidth used for the kernel.
+        """
+        super().__init__()
+        self.parameter_dict['alpha'] = 'exponent'
+        self.parameter_dict['sigma'] = 'bandwidth'
+        if alpha is None:
+            self.raw_alpha = nn.Parameter(torch.empty(1), requires_grad=trainable)
+            self.init_required = True
+        else:
+            self.raw_alpha = nn.Parameter(alpha, requires_grad=trainable)
+            self.init_required = False
+        if sigma is None:
+            self.log_sigma = nn.Parameter(torch.empty(1), requires_grad=trainable)
+            self.init_required = True
+        else:
+            self.log_sigma = nn.Parameter(sigma.log(), requires_grad=trainable)
+            self.init_required = False
+        self.init_fn_alpha = init_fn_alpha
+        self.init_fn_sigma = init_fn_sigma
+        if active_dims is not None:
+            self.active_dims = torch.tensor(active_dims)
+        else:
+            self.active_dims = None
+        self.feature_axis = feature_axis
+        self.trainable = trainable
+
+    @property
+    def alpha(self) -> torch.Tensor:
+        return self.raw_alpha
```

Comment:
Not sure why we need to define `raw_alpha` and this property?

Reply:
Was doing that for some data type consistency, removed as now there is a new parameter implementation.